spynnaker/pyNN/models/neural_projections/connectors/one_to_one_connector.py

```diff
@@ -43,7 +43,9 @@ def get_n_connections_from_pre_vertex_maximum(
 
         slice_min_delay = min(delays)
         slice_max_delay = max(delays)
-        if slice_min_delay >= min_delay and slice_max_delay <= max_delay:
+        if min_delay <= slice_max_delay <= max_delay:
+            return 1
+        if min_delay <= slice_min_delay <= max_delay:
```

Comment:
Right now, this is effectively an “or” that returns 1 if _either_ delay is in range. Should it instead check if _both_ delays are in range? If so, it can become:
```python
if min_delay <= slice_min_delay <= slice_max_delay <= max_delay:
    return 1
```

Reply:
I don't think this is quite right so far - it is meant to be an overlap check.  See for an example: 
https://github.com/SpiNNakerManchester/sPyNNaker/blob/master/spynnaker/pyNN/models/neural_projections/projection_machine_edge.py#L44

Reply:
Obviously the above is checking the overlap of something different, but the idea is similar.

Reply:
It needs to be a separate check for each delay as the values that are passed in for "min_delay" and "max_delay" are different depending on the circumstances (in this case, whether a delay extension is required or not).  The way it's written works properly in that it gives the correct answer in each scenario, but I agree there might be a more intuitive way of writing it.  The problem is that it is an "overlap check", but it's performed over two separate calls to the function...

Reply:
Rewritten to use or, hopefully making it a bit clearer.